jdaviz/core/template_mixin.py

```diff
@@ -1029,14 +1023,23 @@ def _delete_subset(self, subset):
         if self.selected not in self.labels:
             self._apply_default_selection()
 
-    def _update_subset(self, subset, attribute=None):
-        if self._allowed_type is not None and _subset_type(subset) != self._allowed_type:
-            return
+    def _is_valid_item(self, subset):
+        def is_spectral(subset):
+            return _subset_type(subset) == 'spectral'
 
+        def is_spatial(subset):
+            return _subset_type(subset) == 'spatial'
+
+        def is_not_annulus(subset):
```

Comment:
Just so I understand, we still have to update the template itself to add any future custom filters?

Reply:
You can pass the callable directly if you want (in the template python code).  This is just a nice way of having them be re-usable across multiple plugins without having to import them from somewhere.

For example, in a plugin you could have:

```
is_circular = lambda subset: isinstance(subset.subset_state.roi, CircularROI)
self.subset = SubsetSelect(self,
                           'subset_items',
                           'subset_selected',
                           default_text=None,
                           filters=['is_spatial', is_circular])

```

Reply:
I just not sure what else actually needs "is_not_annulus" besides that one dropdown in Aperture Photometry. Should we move this one to the plugin?

Reply:
I think keeping it consolidated in one place is still useful for knowing where to look for how various filters are defined, but don't have a strong preference.